pypass/tests/test_command.py

```diff
@@ -480,12 +479,12 @@ def test_generate_in_place(self):
         self.assertEqual(len(new_password), 10)
         self.assertEqual(remainder, 'second')
 
-    @pypass.tests.skipIfTravis
     def test_generate_clip(self):
         generate = self.run_cli(['generate', '-c', 'clip.me'])
 
         self.assertEqual(generate.output, 'Copied clip.me to clipboard.\n')
 
+        time.sleep(0.1)
```

Comment:
Aww this is required?

timing dependent tests always bite you back.

Is this because pypass exits before waiting for xclip to exit?

Reply:
Unfortunately, I'm not well versed in the inner workings of `xclip`, but if my intuition is correct, invoking `xclip` is async: it terminates the invoked process before its background process clears and writes on (also propably not atomically) the clipboard. Therefore, this test could (and sometimes did before I added `time.sleep`) read the clipboard before the desired content is there. Since there isn't a `Future` I could wait on, I chose this method instead.

What I really don't understand is why this test is problematic, but `test_show_clip()` seems to work like a charm, and they are coded the same way...

I appreciate every pointer in the right direction.